dms.tf

```diff
@@ -24,20 +37,69 @@ resource "aws_dms_replication_subnet_group" "replication-subnet-group" {
 
 # Create a new replication instance
 resource "aws_dms_replication_instance" "replication-instance" {
-  allocated_storage           = 20
+  allocated_storage           = var.allocated_storage
   apply_immediately           = true
-  auto_minor_version_upgrade  = true
-  engine_version              = "3.1.3"
+  auto_minor_version_upgrade  = false
+  engine_version              = "3.4.5"
   publicly_accessible         = false
   replication_instance_class  = "dms.t2.medium"
-  replication_instance_id     = "dms-replication-instance-${random_id.id.hex}"
+  replication_instance_id     = "${var.team_name}-dms-${random_id.id.hex}"
   replication_subnet_group_id = aws_dms_replication_subnet_group.replication-subnet-group.id
 
   tags = {
-    Name        = "${var.team_name} Replication Instance"
-    Description = "Managed by Terraform"
-    Env         = var.environment-name
-    Application = var.application
-    Owner       = var.team_name
+    Name                   = "${var.team_name} Replication Instance"
+    Description            = "Managed by Terraform"
+    Application            = var.application
+    Owner                  = var.team_name
+    is-production          = var.is-production
+    environment-name       = var.environment-name
+    infrastructure-support = var.infrastructure-support
+  }
+}
+
+resource "aws_iam_user" "dms_user" {
+  name = "dms-${var.team_name}-${random_id.id.hex}"
+  path = "/system/dms-user/"
+}
+
+resource "aws_iam_access_key" "dms_key" {
+  user = aws_iam_user.dms_user.name
+}
+
+data "aws_iam_policy_document" "dms_policy" {
+  statement {
+    actions = [
+      "dms:*",
```

Comment:
I'm sure this shouldn't be a wildcard, but there may be something clever you're doing with conditions. Can you clarify?